src/slice.ts

```diff
@@ -35,7 +45,18 @@ export default function create<
     slice,
   });
 
-  const actionMap = actionKeys.reduce<{ [A in keyof Actions]: Actions[A] }>(
+  
+  const actionMap = actionKeys.reduce<
+    {
+      [A in keyof Ax]: Ax extends {[s:string]: {}}
```

Comment:
There's a lot going on here.  To describe what's happening: If `Ax` extends a hash where values are an empty object then a) the the value of the reduce should be a function that maybe accepts payload which is the value of the `Actions` map and returns an action where payload is `Actions` map.  Or it checks to see if `Ax[A]` is null and if so b) it returns a function that returns an `Action`.  c) Or it returns a).

Does that sound right?

So the first check figures out if `Ax` is a hashmap type and if so it returns the `Actions` value for that key.  If not it checks if `Ax[A]` is null if so return a basic action creator if not return `Actions` value for that key.

Reply:
That part is quite tricky, it's actually unneccessary if the user provides an Actions type but due to some Typescript weirdness `Actions = {[key: string]: any}` creates an action creator type of `(p: {}) => Actions` which requires an empty object as an argument so I had to use a round about method to assign `Actions` as `{[key: string]: any}`  if no type is provided.

Basically
`Actions = undefined` sets the Actions type to undefined if not provided.

then the ternary operator
`Actions extends undefined ? {[s:string]:any} : Actions`
is used to check if actions is `undefined` returning `{[key: string]: any}` if it is or Actions if it isn't

I made the ternary into a type now
```js
type NoUndefinedActionsCheck<Actions> = Actions extends undefined ? {[s:string]:any} : Actions
```
I then alias the result of the ternary typecheck to `Ax`

So now we get to 

```js
{
      [A in keyof Ax]: Ax extends {[s:string]: {}}
      ? (payload?: Ax[A]) => Action<Ax[A]>
      :
      Ax[A] extends null
      ? 
      () => Action 
      : (payload: Ax[A]) => Action<Ax[A]>
    }
```
`[A in keyof Ax]:` is obviously used to create a mapped type with the keys of `Ax`

For the ternary, the first operator `Ax extends {[s:string]: {}}` checks if `Ax` is of type `{[s:string]: {}}` , this will only match if `Actions` was `undefined` i.e not provided because `Ax` will be of type `{[s:string]: any}`

Then all the action creator types will be `(payload?: Ax[A]) => Action<Ax[A]>` with optional payloads

*I honestly don't know why this doesn't work if you use `Actions = {[s:string]: any}` to default Actions if it is not given, instead I have to use a round about method.*

If this doesn't trigger, it means `Actions` interface was given and `Ax = Actions` 
The next ternary `Ax[A] extends null`,  `Ax[A]` refers to type of the current `Actions` key, if it is `null` then it indicates that a payload is not required so the action creator type is `() => Action`

If `Ax[A]` is not `null` the action creator is of type ` (payload: Ax[A]) => Action<Ax[A]>` which means it requires a payload of type `Ax[A]`

Short example

```js
interface Actions {
set: number;
reset: null;
}

const { actions } = robodux<number,Actions>({
   //.....
   })

actions.set // <- type:  (payload: number) => Action<number>
                  // an Actions interface was given so the first ternary did not trigger,
                  //  and Ax[A] in this case is not null but number so it falls through to the last check

actions.reset  // <- type:   () => Action 
                      // Ax[A] is null in this instance, triggering the second ternary check

// If no Actions interface was given to robodux 
// both actions.set an actions.reset would be of type:  (payload?: any) => Action<any>
```

I planned to head over to the typescript repository to raise an issue